FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/DemoTasks/DefenderDemoExample.c

```diff
@@ -0,0 +1,684 @@
+                                     DEFENDER_API_LENGTH_JSON_REJECTED( THING_NAME_LENGTH ) );
+    }
+
+    return xStatus;
+}
+/*-----------------------------------------------------------*/
+
+static bool prvUnsubscribeFromDefenderTopics( void )
+{
+    bool xStatus = false;
+
+    xStatus = xUnsubscribeFromTopic( &xMqttContext,
+                                     DEFENDER_API_JSON_ACCEPTED( democonfigTHING_NAME ),
+                                     DEFENDER_API_LENGTH_JSON_ACCEPTED( THING_NAME_LENGTH ) );
+
+    if( xStatus == true )
+    {
+        xStatus = xUnsubscribeFromTopic( &xMqttContext,
+                                         DEFENDER_API_JSON_REJECTED( democonfigTHING_NAME ),
+                                         DEFENDER_API_LENGTH_JSON_REJECTED( THING_NAME_LENGTH ) );
+    }
+
+    return xStatus;
+}
+/*-----------------------------------------------------------*/
+
+static bool prvPublishDeviceMetricsReport( uint32_t reportLength )
+{
+    return xPublishToTopic( &xMqttContext,
+                            DEFENDER_API_JSON_PUBLISH( democonfigTHING_NAME ),
+                            DEFENDER_API_LENGTH_JSON_PUBLISH( THING_NAME_LENGTH ),
+                            &( pcDeviceMetricsJsonReport[ 0 ] ),
+                            reportLength );
+}
+/*-----------------------------------------------------------*/
+
+/*
+ * @brief Create the task that demonstrates the Device Defender library API via a
+ * MQTT mutually authenticated network connection with the AWS IoT broker.
+ */
+void vStartDefenderDemo( void )
+{
+    /* This example uses a single application task, which shows that how to
+     * use Device Defender library to generate and validate AWS IoT Device Defender
+     * MQTT topics, and use the coreMQTT library to communicate with the AWS IoT
+     * Device Defender service. */
+    xTaskCreate( prvDefenderDemoTask,      /* Function that implements the task. */
+                 "DemoTask",               /* Text name for the task - only used for debugging. */
+                 democonfigDEMO_STACKSIZE, /* Size of stack (in words, not bytes) to allocate for the task. */
+                 NULL,                     /* Task parameter - not used in this case. */
+                 tskIDLE_PRIORITY,         /* Task priority, must be between 0 and configMAX_PRIORITIES - 1. */
+                 NULL );                   /* Used to pass out a handle to the created task - not used in this case. */
+}
+/*-----------------------------------------------------------*/
+
+void prvDefenderDemoTask( void * pvParameters )
+{
+    bool xStatus = false;
+    BaseType_t xExitStatus = EXIT_FAILURE;
+    uint32_t ulReportLength = 0, i, ulMqttSessionEstablished = 0;
+
+    /* Remove compiler warnings about unused parameters. */
+    ( void ) pvParameters;
+
+    /* Start with report not received. */
+    xReportStatus = ReportStatusNotReceived;
+
+    /* Set a report Id to be used. */
```

Comment:
As mentioned offline, we should write disclaimer about what is expected of the report ID, what we recommend the customer do, and that we are not production ready.